### PR TITLE
[23.3] Image pull secrets for custom registry

### PR DESCRIPTION
**Pull request by d-venkat**

### CODE REVIEW

**Review by vpugar-digital — APPROVED** [review]

File: templates/_helpers.tpl [hunk 1]
```diff
@@ -279,8 +286,6 @@ Compile all warnings into a single message, and call fail.
 {{- define "release.validateValues" -}}
 {{- $messages := list -}}
 {{- $messages = append $messages (include "release.validateValues.ingress.tls" .) -}}
-{{- $messages = append $messages (include "release.validateValues.keystore.passphrase" .) -}}
-{{- $messages = append $messages (include "release.validateValues.license" .) -}}
```
Comment (vpugar-digital) [inline templates/_helpers.tpl:283]: Don't remove validation, fix it if it is not working correct

File: templates/release-statefulset.yaml [hunk 2]
```diff
@@ -82,9 +82,9 @@ spec:
       {{- end }}
       terminationGracePeriodSeconds: {{ .Values.terminationGracePeriodSeconds }}
       {{- if or .Values.initContainers (and (not .Values.external.db.enabled) .Values.postgresql.install) (and (not .Values.external.mq.enabled) .Values.rabbitmq.install) }}
-      {{- if or .Values.global.imagePullSecrets .Values.image.pullSecrets }}
-      {{ include "common.images.renderPullSecrets" ( dict "images" (list .Values.image) "context" .) | nindent 6 }}
-      {{- end }}
+      {{- if or .Values.global.imagePullSecrets .Values.image.pullSecrets .Values.busyBox.image.pullSecrets }}
+      {{- include "release.imagePullSecrets" . | nindent 6 }}
```
Comment (vpugar-digital) [inline templates/release-statefulset.yaml:86]: If part can go inside "release.imagePullSecrets" function

File: templates/_helpers.tpl [hunk 3]
```diff
@@ -190,7 +197,7 @@ Get the report db URL
         {{ .Values.external.db.report.url }}
     {{- else -}}
         {{- if .Values.postgresql.install -}}
-            jdbc:postgresql://{{ include "postgresql.subchart" . }}:{{ .Values.postgresql.service.port }}/xlr-report-db
+            jdbc:postgresql://{{ include "postgresql.subchart" . }}:{{ .Values.postgresql.primary.service.ports.postgresql }}/xlr-report-db
```
Comment (vpugar-digital) [inline templates/_helpers.tpl:200]: This is not valid change, the state of the branch is not correct

Comment (d-venkat) [reply templates/_helpers.tpl]: Merged all changes from the maintenance branch.

**Review by vpugar-digital — APPROVED** [review]

**Pull request merged**